tests/system/action/base.py

```diff
@@ -280,40 +267,91 @@ def create_user_for_meeting(self, meeting_id: int) -> int:
         )
         return user_id
 
+    @with_database_context
     def set_user_groups(self, user_id: int, group_ids: List[int]) -> None:
         assert isinstance(group_ids, list)
-        partitioned_groups = self._fetch_groups(group_ids)
-        try:
-            user = self.get_model(f"user/{user_id}")
-        except DatastoreException:
-            user = {}
-        new_group_ids = list(
-            set(
-                user.get("group_$_ids", [])
-                + [str(meeting_id) for meeting_id in partitioned_groups.keys()]
-            )
+        groups = self.datastore.get_many(
+            [
+                GetManyRequest(
+                    "group",
+                    group_ids,
+                    ["id", "meeting_id", "user_ids", "meeting_user_ids"],
+                )
+            ],
+            lock_result=False,
+        )["group"]
+        meeting_ids: List[int] = list(set((v["meeting_id"] for v in groups.values())))
+        filtered_result = self.datastore.filter(
+            "meeting_user",
+            FilterOperator("user_id", "=", user_id),
+            ["id", "user_id", "meeting_id", "group_ids"],
+            lock_result=False,
+        )
+        meeting_users: dict[int, dict[str, Any]] = {
+            data["meeting_id"]: dict(data)
+            for data in filtered_result.values()
+            if data["meeting_id"] in meeting_ids
+        }
+        last_meeting_user_id = max(
+            [
+                int(k[1])
+                for key in self.created_fqids
+                if (k := key.split("/"))[0] == "meeting_user"
+            ]
+            or [0]
         )
+        meeting_users_new = {
+            meeting_id: {
+                "id": (last_meeting_user_id := last_meeting_user_id + 1),  # noqa: F841
```

Comment:
You can remove the assignment and the `noqa` if I see it correctly - the newly assigned value is never used

Reply:
done